src/dev/flang/ast/Feature.java

```diff
@@ -1637,7 +1637,12 @@ void typeInference(Resolution res)
           }
         choiceTypeCheckAndInternalFields(res);
 
-        _resultType = resultType();
+        _resultType = resultTypeRaw();
+        if (_resultType == null)
+          {
+            AstErrors.missingResultTypeForField(this);
+            _resultType = Types.t_ERROR;
+          }
```

Comment:
I have the impression that `AstErrors.missingResultTypeForField()` could be called for a non-field feature as well, resulting in a precondition failure in `missingResultTypeForField`.

Also, this ensures that `_resultType != null`, so the if inline 1650 is redundant.